### ISSUE TITLE
[Feature Request] Enable programmatic scrolling

### ISSUE THREAD

**Issue by bigopon** [aurelia/ui-virtualization#104]
**I'm submitting a feature request**

* **Library Version:**
Latest

**Please tell us about your environment:**
* **Operating System:**
Any
* **Node Version:**
Any

* **NPM Version:**
3.8.9

* **JSPM OR Webpack AND Version**
Any

* **Browser:**
all 

* **Current behavior:**
Theres no way to programmatically scroll to an index of the virtualized repeat. I think it can be done manually via setting scrolltop and triggering scroll event but thats not easy.

* **Expected/desired behavior:**
Attribute expose a bindable property to scroll. Use case: when linking state of the scrollable content with url, its not easy to update the content accordingly to the url.

**Edit**
I think with new **primaryProperty** feature, exposing another property and setting current one to primary will not cause breaking changes

**Comment by AStoker** [comment]
@bigopon, do you think you could show me an example of how you want to do this?

**Comment by bigopon** [comment]
Hmm, I think something has changed. I recall when I created this issue, i cannot get `virtual-repeat` to render items correctly, as if there was no scroll event fired when I manually set `scrollTop` of the container. 

But now when opening the demo page of virtual repeat:
- then open the control
- select one of the containers
- change the `scrollTop` to any number

Result is good, virtual repeat got triggered and render items correctly. From here i think it's developer's job to set the scroll top base on their url, not the library. (Back then I expected me to set `scrollTop` and dispatch `scroll`).

Tl;dr; The requested is somehow fulfilled (Not what I really wanted, but it's doable via `scrollTop`).
@AStoker 

**Edit**
What I was doing: a PDF reader, each page can be associated with an url hash, i was trying to scroll virtual repeat by binding, instead of `scrollTop` (or maybe both, but scrollTop didn't work).

**Edit 2**
Just realized what I explained above has nothing to do with the request. To answer your question, it could be something like this:

```html
<ol>
  <div virtual-repeat.for='collection.bind: items; item: item; active-index.bind: vmIndex'>${item.name}</div>
</ol>
```
The `activeIndex` is used to determined index of item in the top of visible view area
The `item: item` is how we define item. And btw, i think `virtual-repeat` doesn't support `ol` correctly from this line: 
https://github.com/aurelia/ui-virtualization/blob/master/src/template-strategy.js#L71

```js
const elementName = element.parentNode.localName === 'ul' ? 'li' : 'div';
// should be const elementName = /[ou]l/.test(element.parentNode.localName) ? 'li' : 'div'
```

**Comment by AStoker** [comment]
:+1: There's still some bugs lingering around (especially with tables), thanks for catching `ol`. 
Do you think you could make a PR for this kind of functionality?

**Comment by bigopon** [comment]
Let me have a look at the whole code first then ill get back to you :smile:

**Comment by bigopon** [comment]
I'm not sure how I would go for this, currently `virtual-repeat` is expected to be used with `for`, which wont let us do something like:
```html
<div virtual-repeat.for=item of items; top-index.bind: vmIndex'>${item.name}</div>
```
@AStoker @EisenbergEffect Am I correct ? If so what should I do ? support it in this form only:

```html
<div virtual-repeat='items.bind: items; local: item; top-index.bind: vmIndex'>${item.name}</div
```

**Comment by AStoker** [comment]
I'm wondering how useful this functionality would really be in the end. We don't want to load up the `virtual-repeat` I think. And since we're using real scroll bars, I'm thinking that the best approach would be to let the user just use the natural browser way of scrolling to an item. A simple helper function could be used to "scroll to an item", as all you have to do is calculate the height of each element, and then multiply that by which index you want to scroll to, and set that to the `scrollTop`.

**Comment by bigopon** [comment]
Closing this issue, as it can be possibly done via binding `scrolltop` to container:
```html
<div scrolltop.one-way="pageNumber * height">
  <page virtual-repeat.for="page of pages"></page>
</div>
```
